x-pack/plugins/cases/public/components/actions/assignees/edit_assignees_flyout.test.tsx

```diff
@@ -10,86 +10,79 @@ import userEvent from '@testing-library/user-event';
   });
 
   it('renders correctly', async () => {
-    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+    appMock.render(<EditAssigneesFlyout {...props} />);
 
-    expect(result.getByTestId('cases-edit-assignees-flyout')).toBeInTheDocument();
-    expect(result.getByTestId('cases-edit-assignees-flyout-title')).toBeInTheDocument();
-    expect(result.getByTestId('cases-edit-assignees-flyout-cancel')).toBeInTheDocument();
-    expect(result.getByTestId('cases-edit-assignees-flyout-submit')).toBeInTheDocument();
-
-    await waitForComponentToUpdate();
+    expect(await screen.findByTestId('cases-edit-assignees-flyout')).toBeInTheDocument();
+    expect(await screen.findByTestId('cases-edit-assignees-flyout-title')).toBeInTheDocument();
+    expect(await screen.findByTestId('cases-edit-assignees-flyout-cancel')).toBeInTheDocument();
+    expect(await screen.findByTestId('cases-edit-assignees-flyout-submit')).toBeInTheDocument();
   });
 
   it('calls onClose when pressing the cancel button', async () => {
-    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+    appMock.render(<EditAssigneesFlyout {...props} />);
 
-    userEvent.click(result.getByTestId('cases-edit-assignees-flyout-cancel'));
-    expect(props.onClose).toHaveBeenCalled();
+    userEvent.click(await screen.findByTestId('cases-edit-assignees-flyout-cancel'));
 
-    await waitForComponentToUpdate();
+    await waitFor(() => {
+      expect(props.onClose).toHaveBeenCalled();
+    });
   });
 
   it('calls onSaveAssignees when pressing the save selection button', async () => {
-    const result = appMock.render(<EditAssigneesFlyout {...props} />);
+    appMock.render(<EditAssigneesFlyout {...props} />);
 
-    await waitForComponentToUpdate();
+    expect(await screen.findByText('Damaged Raccoon')).toBeInTheDocument();
 
-    await waitFor(() => {
-      expect(result.getByText('Damaged Raccoon')).toBeInTheDocument();
-    });
-
-    userEvent.click(result.getByText('Damaged Raccoon'));
-    userEvent.click(result.getByTestId('cases-edit-assignees-flyout-submit'));
+    userEvent.click(await screen.findByText('Damaged Raccoon'));
+    userEvent.click(await screen.findByTestId('cases-edit-assignees-flyout-submit'));
 
-    expect(props.onSaveAssignees).toHaveBeenCalledWith({
-      selectedItems: [],
-      unSelectedItems: ['u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0'],
+    await waitFor(() => {
+      expect(props.onSaveAssignees).toHaveBeenCalledWith({
+        selectedItems: [],
+        unSelectedItems: ['u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0'],
+      });
     });
   });
 
   it('shows the case title when selecting one case', async () => {
-    const result = appMock.render(<EditAssigneesFlyout {...props} />);
-
-    expect(result.getByText(basicCase.title)).toBeInTheDocument();
+    appMock.render(<EditAssigneesFlyout {...props} />);
 
-    await waitForComponentToUpdate();
```

Comment:
What do you think about any unresolved promises? (here and above)

I tried running these tests with the option `--detectOpenHandles` but that never worked properly for me. Could you try too?

Reply:
Sorry, I had the auto-merge enabled. Let's talk about it offline. I am interested to hear more about it.